lib/active_replicas/rails4/process_local_connection_handler.rb

```diff
@@ -33,10 +28,23 @@ def connection_pool_list
 
       def establish_connection(owner, spec)
         prefix = '[ActiveReplicas::Rails4::ConnectionHandler#establish_connection]'
-        ActiveRecord::Base.logger&.warn "#{prefix} Ignoring spec for #{owner.inspect}: #{spec.inspect}"
-        ActiveRecord::Base.logger&.info "#{prefix} Called from:\n" + Kernel.caller.first(5).map {|t| "  #{t}" }.join("\n")
+        details = "#{spec.config.inspect} (owner: #{owner.inspect})"
 
-        current_proxy
+        synchronize do
```

Comment:
Satisfy my curiosity: how come you don't need a token for this `synchronize` call?